django_cloud_deploy/workflow/_service_account.py

```diff
@@ -16,12 +16,58 @@
 
 import json
 import os
-from typing import List
+from typing import Any, Dict, List
 
 from django_cloud_deploy.cloudlib import service_account
 
 from google.auth import credentials
 
+KEY_PATH_TEMPLATE = '{}/{}.json'
+
+
+class ServiceAccount(object):
+    """ Service Accounts for permissions in Google Cloud.
+
+    Attributes:
+        id_: Id of your service account. For example, a
+            service account should be in the following format:
+            <service_account_id>@<project_id>.iam.gserviceaccount.com
+        name: Display name of your service account.
+        file_name: Name of the secrets file.
+        roles: Roles the service account should have. Valid roles can be
+            found on https://cloud.google.com/iam/docs/understanding-roles
+        output_path: The path to store your key.
+    """
+    def __init__(self,
+                 id_: str,
+                 name: str,
+                 file_name: str,
+                 roles: List[str],
+                 output_path: str = KEY_PATH_TEMPLATE):
+        """
+
+        Args:
+            id_: Id of your service account. For example, a
+                service account should be in the following format:
+                <service_account_id>@<project_id>.iam.gserviceaccount.com
+            name: Display name of your service account.
+            file_name: Name of the secrets file.
+            roles: Roles the service account should have. Valid roles can be
+                found on https://cloud.google.com/iam/docs/understanding-roles
+            output_path: The path to store your key.
+
+        Returns:
+            service_account: ServiceAccount
+        """
+        self.id_ = id_
+        self.name = name
+        self.file_name = file_name
+        self.roles = roles
+        if output_path == KEY_PATH_TEMPLATE:
```

Comment:
output_path equals to a template is misleading. We can set the default value of output_path to None. Then we can write:

if not output_path:
    output_path =  KEY_PATH_TEMPLATE.format(os.path.expanduser('~'), id_)